### DISCUSSION TITLE
ch3-asm/ch3-01-basic

### DISCUSSION THREAD

**Discussion by giscus** [chai2010/advanced-go-programming-book#629] [General]
# ch3-asm/ch3-01-basic

Go语言高级编程

https://chai2010.cn/advanced-go-programming-book/ch3-asm/ch3-01-basic.html

**Comment by fire1220** [comment]
咱么是用go什么版本的？
我本地版本是1.16.14 执行汇编是报错，提示runtime·printstring和 runtime·printnl没有定义
怎么才能不报错呢？
报错：
main.main: relocation target runtime.printstring not defined for ABI0 (but is defined for ABIInternal)
main.main: relocation target runtime.printnl not defined for ABI0 (but is defined for ABIInternal)
汇编代码：
TEXT ·main(SB), $16-0
    MOVQ ·helloworld+0(SB), AX; MOVQ AX, 0(SP)
    MOVQ ·helloworld+8(SB), BX; MOVQ BX, 8(SP)
    CALL runtime·printstring(SB)
    CALL runtime·printnl(SB)
    RET

**Reply by baipangbai** [reply]
https://github.com/chai2010/advanced-go-programming-book/issues/427


这里有解决方案

**Comment by cod-code-go** [comment]
如果是m2芯片的mac，写的汇编文件应该命名为XX_arm64.s吗？
